parser-typechecker/src/Unison/PrettyPrintEnv.hs

```diff
@@ -20,18 +28,25 @@ import qualified Unison.HashQualified' as HQ'
 import Unison.LabeledDependency (LabeledDependency)
 import qualified Unison.LabeledDependency as LD
 import Unison.Name (Name)
-import Unison.Prelude
+import qualified Unison.Name as Name
+import Unison.Prelude hiding (empty)
 import Unison.Reference (Reference)
 import Unison.Referent (Referent)
 import qualified Unison.Referent as Referent
 
 data PrettyPrintEnv = PrettyPrintEnv
-  { -- names for terms, constructors, and requests
-    terms :: Referent -> Maybe (HQ'.HashQualified Name),
-    -- names for types
-    types :: Reference -> Maybe (HQ'.HashQualified Name)
+  { -- names for terms, constructors, and requests; e.g. [(original name, relativized and/or suffixified pretty name)]
+    termNames :: Referent -> [(HQ'.HashQualified Name, HQ'.HashQualified Name)],
+    -- names for types; e.g. [(original name, relativized and/or suffixified pretty name)]
+    typeNames :: Reference -> [(HQ'.HashQualified Name, HQ'.HashQualified Name)]
   }
 
+terms :: PrettyPrintEnv -> Referent -> Maybe (HQ'.HashQualified Name)
```

Comment:
We re-export versions of the old fields for back-compatibility; these select the current 'top-priority' name for a reference.